roles/pulp_common/tasks/prepare_signing_service.yml

```diff
@@ -0,0 +1,93 @@
+---
+- name: Check if galaxy signing service gpg key exists
+  stat:
+    path: "{{ pulp_certs_dir }}/galaxy_signing_service.gpg"
+  register: stat_galaxy_signing_service
+
+- name: Generate galaxy signing service gpg key  # noqa no-changed-when
+  shell: |
+    cat >gpg.txt <<EOF
+    %echo Generating a basic OpenPGP key
+    Key-Type: default
+    Key-Length: 4096
+    Subkey-Type: default
+    Subkey-Length: default
+    Name-Real: Joe Tester
+    Name-Comment: with no passphrase
+    Name-Email: joe@foo.bar
+    Expire-Date: 0
+    %no-ask-passphrase
+    %no-protection
+    # Do a commit here, so that we can later print "done" :-)
+    %commit
+    %echo done
+    EOF
+    gpg --batch --gen-key gpg.txt
+    gpg --output {{ pulp_certs_dir }}/galaxy_signing_service.gpg --armor --export-secret-key
+    gpg --output {{ pulp_certs_dir }}/galaxy_signing_service.asc --armor --export
+    rm -rf ~/.gnupg
```

Comment:
So if we do this, it will blow away gpg keys (and export all of them) from '{{ ansible_user }}'.

If we're confident no users will set this variable accidentally, we can do this.

Reply:
I'd prefer to run these commands with GNUPG_HOME pointing to a tmpdir then.